aiohttp/web_reqrep.py

```diff
@@ -321,6 +321,15 @@ def json(self, *, loads=json.loads, loader=None):
         return loads(body)
 
     @asyncio.coroutine
+    def multipart(self, *, reader=multipart.MultipartReader):
+        """Return async iterator to process BODY as multipart."""
+        if not self.content_type.startswith('multipart/'):
+            warnings.warn('Attempt to process multipart request with'
```

Comment:
No, that's bad idea since content-type validation must be done before any read method call. Raising exception means we have double ctype validation. Also, people may use custom content-type for their requests like `applicaiton/vnd+multipart` or whatever which won't pass this check.

The reason of this warning is to notify that things may go wrong, but since you call that method we think that you know what are you doing. May be it's practically useless though.